common/txmgr/txmgr.go

```diff
@@ -654,8 +654,8 @@ func (b *Txm[CHAIN_ID, HEAD, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE]) GetTransac
 	}
 	switch tx.State {
 	case TxUnconfirmed, TxConfirmedMissingReceipt:
-		// Return unconfirmed for ConfirmedMissingReceipt since a receipt is required to determine if it is finalized
-		return commontypes.Unconfirmed, nil
+		// Return pending for ConfirmedMissingReceipt since a receipt is required to determine if it is finalized
```

Comment:
Sorry to nit but now that there's a state after pending maybe something like this is more accurate
```suggestion
		// Return pending for ConfirmedMissingReceipt since a receipt is required to consider it as unconfirmed
```